src/vpk/Velopack.Packaging.Windows/Commands/WindowsPackCommandRunner.cs

```diff
@@ -261,16 +263,60 @@ private void SignFilesImpl(WindowsSigningOptions options, Action<int> progress,
         // signtool.exe does not work if we're not on windows.
         if (!VelopackRuntimeInfo.IsWindows) return;
 
-        if (!string.IsNullOrEmpty(signParams)) {
+        if (!string.IsNullOrEmpty(trustedSignMetadataPath)) {
+            Log.Info($"Use Azure Trusted Signing service for code signing. Metadata file path: {trustedSignMetadataPath}");
+
+            string dlibPath = await GetDlibPath(CancellationToken.None);
+            signParams = $"/fd SHA256 /tr \"http://timestamp.acs.microsoft.com\" /v /debug /td SHA256 /dlib \"{dlibPath}\" /dmdf \"{trustedSignMetadataPath}\"";
+            helper.Sign(filePaths, signParams, signParallel, progress, false);
+        } else if (!string.IsNullOrEmpty(signParams)) {
             helper.Sign(filePaths, signParams, signParallel, progress, false);
         }
     }
 
+    [SupportedOSPlatform("windows")]
+    private async Task<string> GetDlibPath(CancellationToken cancellationToken)
+    {
+        // DLib library is required for Azure Trusted Signing. It must be in the same directory as SignTool.exe.
+        // https://learn.microsoft.com/azure/trusted-signing/how-to-signing-integrations#download-and-install-the-trusted-signing-dlib-package
+        var signToolPath = HelperFile.SignToolPath;
+        var signToolDirectory = Path.GetDirectoryName(signToolPath);
+        var dlibPath = Path.Combine(signToolDirectory, HelperFile.AzureDlibFileName);
+        if (File.Exists(dlibPath)) {
+            return dlibPath;
+        }
+        Log.Info($"Downloading Azure Trusted Signing dlib to '{dlibPath}'");
+        var dl = new NuGetDownloader();
+
+        using MemoryStream nupkgStream = new();
+        await dl.DownloadPackageToStream("Microsoft.Trusted.Signing.Client", "1.*", nupkgStream, cancellationToken);
+
+        nupkgStream.Position = 0;
+
+        string parentDir = NugetUtil.BinDirectory + Path.AltDirectorySeparatorChar;
+        if (Environment.Is64BitOperatingSystem) {
+            parentDir += "x64";
+        } else {
+            parentDir += "x86";
```

Comment:
Shouldn't ever extract x86 because we don't ship an x86 signtool that can use it